File: C/CS120/p1_quadratic.c
/* Roots of quadratic equation ax2+bx+c=0 */

#include <stdio.h>
#include <math.h>

void main()
{
	int a, b, c, d;
	float root1, root2, real, complex;

	printf("Enter a, b and c: \n");
	scanf("%d%d%d", &a, &b, &c);
	d = b*b - 4*a*c;

	if (d > 0)
	{
		root1 = (float)(-b + sqrt(d))/(2*a);
		root2 = (float)(-b - sqrt(d))/(2*a);
		printf("Roots are %f and %f\n", root1, root2);
	}

	else if (d == 0)
	{
		root1 = root2 = ((float)(-b + sqrt(d)))/(2*a);
		printf("Root of the equation is %f\n", root1);
	}

	else if(d < 0)
	{
		real = (float)(-b)/(2*a);
		complex = (float)(sqrt(-d)/(2*a));
		printf("Roots are %f + i%f and %f - i%f\n", real, complex, real, complex);
	}
}

File: Python/CS110/p1_quadratic.py
# To find the roots of a quadratic equation

a = int(input("Enter a: "))
b = int(input("Enter b: "))
c = int(input("Enter c: "))

d = b ** 2 - 4 * a * c

if d >= 0:
	root1 = (-b + d ** 0.5) / (2 * a)
	root2 = (-b - d ** 0.5) / (2 * a)
	print("Roots are {0} and {1}".format(root1, root2))

else:
	x = -b / (2 * a)
	y = (-d) ** (0.5) / (2 * a)
	print("Roots are {0} + i{1} and {0} - i{1}".format(x, y))
